src/components/select/bl-select.ts

```diff
@@ -661,17 +661,26 @@ export default class BlSelect<ValueType extends FormValue = string> extends Form
     this._handleSearchEvent();
 
     this._connectedOptions.forEach(option => {
-      const isVisible = option.textContent
-        ?.toLocaleLowerCase(this.userLang)
-        .includes(this._searchText.toLocaleLowerCase(this.userLang));
+      const searchText = this._searchText.toLowerCase();
+      let optionText = "";
+
+      if (option.textContent) {
+        try {
+          // Try locale-specific comparison first
+          optionText = option.textContent.toLocaleLowerCase(this.userLang);
+        } catch {
+          // Fallback to basic toLowerCase if locale is not supported
+          optionText = option.textContent.toLowerCase();
+        }
+      }
```

Comment:
why we're using try-catch here?

Reply:
It’s for code coverage. While I can't see the exact problem, if we don’t address it, the tests might behave unpredictably